Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp

```diff
@@ -111,11 +112,23 @@ namespace Camera
             beginInsertRows(QModelIndex(), rowCount(), rowCount());
             m_cameraItems.push_back(cameraId);
             endInsertRows();
+
+            if (m_lastActiveCamera.IsValid() && m_lastActiveCamera == cameraId)
+            {
+                Camera::CameraRequestBus::Event(cameraId, &Camera::CameraRequestBus::Events::MakeActiveView);
+            }
         }
 
         void CameraListModel::OnCameraRemoved(const AZ::EntityId& cameraId)
         {
-            auto cameraIt = AZStd::find_if(m_cameraItems.begin(), m_cameraItems.end(),
+            if (m_firstEntry)
+            {
+                CameraSystemRequestBus::BroadcastResult(m_lastActiveCamera, &CameraSystemRequestBus::Events::GetActiveCamera);
```

Comment:
suggestion: Instead of using `m_firstEntry` can we not just use `m_lastActiveCamera.IsValid()` to see if it's been set or not? Is that essentially what it's being used for? (related to other comment)

Reply:
The issue here is that since all the cameras get removed this value is getting updated so for example, we have 4 cameras
-Camera 1 - ID = 1
-Camera 2 - ID = 2
-Camera 3 - ID = 3
-Camera 4 - ID = 4  -  current active camera

When the camera gets removed `m_lastActiveCamera == 4` but then at this point the currentView is changed to one of the available cameras (let's say Camera 3) so when it removes the next camera then `m_lastActiveCamera' will be 3 (which is not the value we want, we want 4) so actually `m_lastActiveCamera` will be a valid EntityID but will not be the correct one. So with this `m_firstEntry` value we ensure that we will get the active camera until we add a new one.
Hope that makes sense